bus-mapping/src/circuit_input_builder/input_state_ref.rs

```diff
@@ -1708,11 +1737,14 @@ impl<'a> CircuitInputStateRef<'a> {
         // memory word writes to destination word
         for chunk in code_slot_bytes.chunks(32) {
```

Comment:
To get the `value_prev` into the `copy_steps`, we could merge these two loops like this:

```rust
for (indexes, chunk) in code_slot_bytes.enumerate().chunks(32) {
    let prev_bytes = …memory write…
    …
    for ((idx, value), value_prev) in indexes.zip(chunk).zip(prev_bytes) {
        let address = dst_begin_slot + idx;
        …
```

Reply:
It’s nice because the data is processed once and it’s clear that the two actions are connected: write memory AND generate copy events for the same data.